src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterField.hpp

```diff
@@ -246,6 +246,14 @@ struct DDSFilterField final : public DDSFilterValue
                     float_value = data->get_float128_value(member_id);
                     break;
 
+                case EK_COMPLETE:
+                {
+                    uint32_t enum_value;
+                    ret = !!data->get_enum_value(enum_value, member_id);
```

Comment:
`get_enum_value` returns a `ReturnCode_t` which if if fails returns a error code different from `0` and if it is successful, returns `0` (`RETCODE_OK`). Negating twice the return code is done to convert to a binary result. However, even though all error codes are transformed into `1` and the successful code keeps its value of `0`, when using that code as a boolean variable the return code loses its proper meaning. Right now, `ret` is `1` (`true`) when there is an error code and `0` (`false`) when the method returns successfully. I think that here it is enough negating only once.

Do we need a regression test?

Reply:
`ReturnCode_t` has an overload of the `!` operator that returns `value_ != 0`. See [here](https://github.com/eProsima/Fast-DDS/blob/4a3f9194f3393721b1452284de325e0c4339933e/include/fastrtps/types/TypesBase.h#L242).